platforms/jvm/language-java/src/integTest/groovy/org/gradle/api/tasks/compile/JavaCompileProblemsIntegrationTest.groovy

```diff
@@ -14,18 +14,29 @@
  * limitations under the License.
  */
 
-package org.gradle.api.tasks.compile;
+package org.gradle.api.tasks.compile
 
 import org.gradle.integtests.fixtures.AbstractIntegrationSpec
-import spock.lang.Ignore;
+import org.gradle.integtests.fixtures.problems.ReceivedProblem
+import org.gradle.test.fixtures.file.TestFile
+import spock.lang.Issue
 
 /**
  * Test class verifying the integration between the {@code JavaCompile} and the {@code Problems} service.
  */
-@Ignore
 class JavaCompileProblemsIntegrationTest extends AbstractIntegrationSpec {
 
+    // The `assertProblem` method will store visited files here, and `assertAllFilesVisited` will check it
+    List<String> assertedFileLocations = []
+
     def setup() {
+        enableProblemsApiCheck()
+
+        propertiesFile << """
+            # Feature flag as of 8.6 to enable the Problems API
+            org.gradle.compile.use-problems-api=true
```

Comment:
Same as above. We'd need a feature flag when your changes would have a major impact by default.

Reply:
But isn't this a feature flag for you? 
Would you like to introduce a new `--enable-compilation-problems` flag?

Reply:
https://github.com/gradle/gradle/pull/27539/files#r1452337507